extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../../../src/services';
+const ALLOWED_AD_PROVIDER = 'sr';
+import {createElementWithAttributes} from '../../../../../src/dom';
+import {dict} from '../../../../../src/utils/object';
+import {getValueForExpr} from '../../../../../src/json';
+
+/**
+ * @param {!JsonObject} media
+ * @param {AmpElement} apesterElement
+ * @param {!JsonObject} consentObj
+ */
+export function handleCompanionVideo(media, apesterElement, consentObj) {
+  const companionCampaignOptions = getValueForExpr(
+    /**@type {!JsonObject}*/ (media),
+    'campaignData.companionCampaignOptions'
+  );
+  const videoSettings = getValueForExpr(
+    /**@type {!JsonObject}*/ (media),
+    'campaignData.companionOptions.video'
+  );
+  const position = getCompanionPosition(
+    /**@type {!JsonObject}*/ (videoSettings)
+  );
+
+  if (
+    !companionCampaignOptions ||
+    !videoSettings ||
+    !videoSettings['enabled'] ||
+    videoSettings['provider'] !== ALLOWED_AD_PROVIDER ||
+    !position ||
+    position === 'floating'
+  ) {
+    return;
+  }
+  const macros = getSrMacros(
+    media,
+    companionCampaignOptions['companionCampaignId'],
+    apesterElement,
+    consentObj
+  );
+  constructCompanionSrElement(
+    videoSettings['videoTag'],
+    position,
+    /** @type {!JsonObject} */ (macros),
+    apesterElement
+  );
+}
+
+/**
+ * @param {!JsonObject} video
+ * @return {?string}
+ */
+function getCompanionPosition(video) {
+  if (!video) {
+    return null;
+  }
+  if (video['companion']['enabled']) {
+    return 'above';
+  }
+  if (video['companion_below']['enabled']) {
+    return 'below';
+  }
+  if (video['floating']['enabled']) {
+    return 'floating';
+  }
+  return null;
+}
+
+/**
+ * @param {string} videoTag
+ * @param {string} position
+ * @param {!JsonObject} macros
+ * @param {AmpElement} apesterElement
+ */
+function constructCompanionSrElement(
```

Comment:
Consider renaming this to something like `addCompanionSrElement`, since it is doing more than just constructing one. Also, what is `Sr` in this case? Can you either expand it out or add a description to the method comments?

Reply:
Thanks, I changed it to addCompanionSrElement. Sr is a common phrase in Apester it's mean Streamrail video.